js/util/proxy.js

```diff
@@ -0,0 +1,75 @@
+if (typeof require !== 'undefined') {
+  var settings = require('util/settings/settings.js')
+}
+const { BrowserWindow } = require('electron')
+const { session } = require('electron')
+
+var proxy = {
+  rules: {
+    proxyBypassRules: "localhost"
+  },
+  initialize: function () {
+    settings.listen("proxyMode", function(value){
+      if (value == 'no-proxy') {
+        proxy.changeHandler("")
+      }
+      else {
+        var proxyRules = {}
+        settings.listen('proxies', function(proxy){
+          if ( proxy.httpHost ) {

```

Comment:
`proxy` could be undefined here if the setting has never been modified, so there needs to be a check for that.